nilearn/mass_univariate/permuted_least_squares.py

```diff
@@ -611,18 +614,290 @@ def permuted_ols(
+        "logp_max_t": -np.log10(vfwe_pvals),
+        "h0_max_t": vfwe_h0,
+    }
+
+    if tfce:
+        # We can use the same approach for TFCE that we use for vFWE
+        h0_tfcemax = np.hstack(h0_tfce_parts)
+        tfce_scores_as_ranks = np.zeros((n_regressors, n_descriptors))
+        for tfce_scores_as_ranks_part in tfce_scores_as_ranks_parts:
+            tfce_scores_as_ranks += tfce_scores_as_ranks_part
+
+        tfce_pvals = (n_perm + 1 - tfce_scores_as_ranks) / float(1 + n_perm)
+        neg_log10_tfce_pvals = -np.log10(tfce_pvals)
+
+        outputs["tfce"] = tfce_original_data.T
+        outputs["logp_max_tfce"] = neg_log10_tfce_pvals
+        outputs["h0_max_tfce"] = h0_tfcemax
+
+    if threshold is not None:
+        cluster_stats = _compute_cluster_level_statistics(
+            csfwe_h0_parts,
+            cmfwe_h0_parts,
+            vfwe_pvals,
+            masker,
+            scores_original_data,
+            n_regressors,
+            threshold_t,
+            two_sided_test,
+        )
+
+        outputs["size"] = cluster_stats["size"]
+        outputs["logp_max_size"] = -np.log10(cluster_stats["size_pvals"])
+        outputs["h0_max_size"] = cluster_stats["size_h0"]
+        outputs["mass"] = cluster_stats["mass"]
+        outputs["logp_max_mass"] = -np.log10(cluster_stats["mass_pvals"])
+        outputs["h0_max_mass"] = cluster_stats["mass_h0"]
+
+    return outputs
+
+
+def _check_args_permuted_ols(
+    target_vars,
+    tested_vars,
+    confounding_vars,
+    n_jobs,
+    output_type,
+    tfce,
+    threshold,
+    masker,
+):
+    """Validate input arguments for permuted_ols.
+
+    Parameters
+    ----------
+    tested_vars : array-like, shape=(n_samples, n_regressors)
+        Explanatory variates, fitted and tested independently from each others.
+
+    target_vars : array-like, shape=(n_samples, n_descriptors)
+        fMRI data to analyze according to the explanatory and confounding
+        variates.
+
+    confounding_vars : array-like, shape=(n_samples, n_covars), optional
+        Confounding variates (covariates), fitted but not tested.
+
+    n_jobs : :obj:`int`
+        Number of parallel workers.
+        If -1 is provided, all CPUs are used.
+        A negative number indicates that all the CPUs except (abs(n_jobs) - 1)
+
+    output_type : {'legacy', 'dict'}, optional
+        Determines how outputs should be returned.
+
+    tfce : :obj:`bool`
+        Whether to calculate :term:`TFCE` as part of the permutation procedure
+        or not.
+
+    threshold : None or :obj:`float`
+        Cluster-forming threshold in p-scale.
+
+    masker : None or :class:`~nilearn.maskers.NiftiMasker` or \
+            :class:`~nilearn.maskers.MultiNiftiMasker`
+        A mask to be used on the data.
+        This is required for cluster-level inference, so it must be provided
+        if ``threshold`` is not None.
+
+    """
     if n_jobs == 0:  # invalid according to joblib's conventions
```

Comment:
will double check and remove if we can

note to self: check if similar checks do not exist in the rest of the codebase

Reply:
so it turns out this check is also there to prevent an eventual dividing by zero error that happens when computing the how to chunk the permutations.

```python
    if n_perm > n_jobs:
        n_perm_chunks = np.asarray([n_perm / n_jobs] * n_jobs, dtype=int)
        n_perm_chunks[-1] += n_perm % n_jobs
```